uaclient/entitlements/repo.py

```diff
@@ -68,21 +67,20 @@ def enable(self, *, silent_if_inapplicable: bool = False) -> bool:
         if not self.can_enable(silent=silent_if_inapplicable):
             return False
         if not self.setup_apt_config():
+            self.remove_apt_config()  # Cleanup any apt sources/auth artifacts
             return False
         if self.packages:
             try:
                 print(
                     'Installing {title} packages'.format(title=self.title))
                 for msg in self.messaging.get('pre_install', []):
                     print(msg)
-                util.subp(
+                apt.run_apt_command(
                     ['apt-get', 'install', '--assume-yes'] + self.packages,
-                    capture=True, retry_sleeps=APT_RETRIES)
-            except util.ProcessExecutionError:
-                self._cleanup()
-                logging.error(
                     status.MESSAGE_ENABLED_FAILED_TMPL.format(
                         title=self.title))
+            except (util.ProcessExecutionError, exceptions.UserFacingError):
```

Comment:
We actually still display the error because capture=True is provided, which then gives us a console error log  like

ERROR: Failed running command 'apt-get install --assume-yes idontexist' [exit(100)]. Message: b'E: Unable to locate package idontexist\n'

Maybe this is undesired behavior of subp(capture=True) that I introduced a while back on subp refactor. If we fail a subp and capture=True was provided subp also logging.errors the error before raising the CalledProcessError. this is probable the wrong behavior now that you've brought it up about catching the UserFacingError :/

Reply:
I'm not 100% sure, but I think allowing the `UserFacingError` to bubble out (with cleanup moved to a `finally:` clause, I guess?) and calling `apt.run_apt_command` inside a `with disable_log_to_console():` might be what we want here?

Reply:
I took another cut at it that we can review on Monday. I've moved the cleanup inside setup_apt_config to only be called remove_apt_config once we've actually left breadcrumbs of apt config setup. that we we aren't unnecessarily calling remote_apt_config unless we need to.  I also dropped the ProcessExecutionError from the except because run_apt_command doesn't actually raise that error type. We re-raise UserFacingError now after performing a remote_apt_config